subsys/debug/tracing/CMakeLists.txt

```diff
@@ -1,6 +1,10 @@
 # SPDX-License-Identifier: Apache-2.0
 
-zephyr_include_directories_ifdef(CONFIG_TRACING include)
+zephyr_include_directories_ifdef(CONFIG_TRACING
+  include
+  ${ZEPHYR_BASE}/kernel/include
+  ${ZEPHYR_BASE}/arch/${ARCH}/include
+  )
```

Comment:
s/zephyr_include/zephyr_library_include/ ?

Reply:
good catch. will fix.

Reply:
So it turns out that one `zephyr_include` actually hid many other components that include the private headers; in particular, the SoCs (almost all soc/**/CMakeLists.txt use zephyr_sources, instead of zephyr_library_sources).

I think we should add the arch and kernel private include paths to soc/ too. After all, 'soc' can be thought of as an extension to 'arch' and will need to access a lot of private 'arch' stuff.

@andrewboie What do you think about this?